db/migration/1688372371221-AddPresentationToVariables.ts

```diff
@@ -0,0 +1,91 @@
+import { MigrationInterface, QueryRunner } from "typeorm"
+
+export class AddPresentationToVariables1688372371221
+    implements MigrationInterface
+{
+    public async up(queryRunner: QueryRunner): Promise<void> {
+        await queryRunner.query(
+            `ALTER TABLE variables
+                ADD COLUMN schemaVersion INT NOT NULL DEFAULT 1,
+                ADD COLUMN processingLevel ENUM('minor', 'major'),
+                ADD COLUMN processingLog JSON,
+                ADD COLUMN titlePublic TEXT,
+                ADD COLUMN titleVariant TEXT,
+                ADD COLUMN producerShort TEXT,
+                ADD COLUMN citationInline TEXT,
+                ADD COLUMN descriptionShort TEXT,
+                ADD COLUMN descriptionFromProducer TEXT,
+                -- ADD COLUMN topicTagsLinks JSON,
+                ADD COLUMN keyInfoText JSON,
+                ADD COLUMN processingInfo TEXT,
+                ADD COLUMN licenses JSON,
+                ADD COLUMN presentationLicense JSON,
+                ADD COLUMN grapherConfigETL JSON;`
+        )
+        await queryRunner.query(
+            `ALTER TABLE datasets
+                ADD COLUMN updatePeriod INT;`
+        )
+
+        // make variables.souceId nullable
+        await queryRunner.query(
+            `ALTER TABLE variables MODIFY sourceId int NULL;`
+        )
+
+        await queryRunner.query(`
+        CREATE TABLE posts_gdocs_variables_faqs (
+            gdocId varchar(255) CHARACTER SET utf8mb4 COLLATE utf8mb4_0900_as_cs NOT NULL,
```

Comment:
Any reason for the special charset and collation? (leave a one-line comment perhaps)

Reply:
This is a foreign key to `posts_gdocs.id` which has this charset and collation (no idea why). I'll add a comment.

Reply:
I don't know when or how exactly this happened but our database has the default collation to do case INsensitive comparisons but then most (all?) tables define it as case sensitive which is why we have this explicitly in quite a few places.

Reply:
I first encountered this when wanting to create a FK for `posts_gdocs` - gdoc IDs _are_ case-sensitive but I don't think that would affect the operations we do with that table, either way.

@danyx23 `UpdateCharaterSet1647947484386` collated everything to `COLLATE utf8mb4_0900_as_cs` - just for my own understanding, does that concur with what you're saying about our DB's default collation?